src/test/kotlin/no/nav/syfo/melding/status/kafka/KafkaDialogmeldingStatusConsumerSpek.kt

```diff
@@ -0,0 +1,123 @@
+package no.nav.syfo.melding.status.kafka
+
+import io.ktor.server.testing.*
+import io.mockk.*
+import no.nav.syfo.melding.api.toMeldingTilBehandler
+import no.nav.syfo.melding.database.createMeldingTilBehandler
+import no.nav.syfo.melding.status.MeldingStatusService
+import no.nav.syfo.melding.status.database.createMeldingStatus
+import no.nav.syfo.melding.status.domain.MeldingStatus
+import no.nav.syfo.melding.status.domain.MeldingStatusType
+import no.nav.syfo.testhelper.*
+import no.nav.syfo.testhelper.generator.generateKafkaDialogmeldingStatusDTO
+import no.nav.syfo.testhelper.generator.generateMeldingTilBehandlerRequestDTO
+import no.nav.syfo.testhelper.mock.mockKafkaConsumer
+import org.amshove.kluent.shouldBeEqualTo
+import org.amshove.kluent.shouldBeGreaterThan
+import org.spekframework.spek2.Spek
+import org.spekframework.spek2.style.specification.describe
+import java.util.*
+
+val meldingTilBehandler = generateMeldingTilBehandlerRequestDTO().toMeldingTilBehandler(
+    personident = UserConstants.ARBEIDSTAKER_PERSONIDENT,
+)
+
+class KafkaDialogmeldingStatusConsumerSpek : Spek({
+    with(TestApplicationEngine()) {
+        start()
+        val externalMockEnvironment = ExternalMockEnvironment.instance
+        val database = externalMockEnvironment.database
+
+        val kafkaDialogmeldingStatusConsumer = KafkaDialogmeldingStatusConsumer(
+            database = database,
+            meldingStatusService = MeldingStatusService(),
+        )
+
+        afterEachTest {
+            database.dropData()
+        }
+
+        describe("${KafkaDialogmeldingStatusConsumer::class.java.simpleName}: pollAndProcessRecords") {
+            it("Creates no melding-status when no melding in db") {
+                val kafkaDialogmeldingStatusDTO = generateKafkaDialogmeldingStatusDTO(
+                    meldingUUID = UUID.randomUUID(),
+                    status = MeldingStatusType.OK,
+                )
+                val mockConsumer = mockKafkaConsumer(kafkaDialogmeldingStatusDTO, DIALOGMELDING_STATUS_TOPIC)
+
+                kafkaDialogmeldingStatusConsumer.pollAndProcessRecords(mockConsumer)
+
+                verify(exactly = 1) { mockConsumer.commitSync() }
+
+                database.getMeldingStatus().size shouldBeEqualTo 0
+            }
+            it("Creates no melding-status for unknown melding") {
+                database.connection.use {
+                    it.createMeldingTilBehandler(meldingTilBehandler)
+                }
```

Comment:
Finnes en `createMeldingerTilBehandler` på `DatabaseInterface`-nivå i `TestDatabase.kt` også, hvis man vil slippe connection.

Reply:
Ja, så det, men den setter `uuid = UUID.randomUUID()` så da måtte jeg gjort noen tilpasninger der for å få uuiden tilbake. Da var det like greit å gjøre det sånn :)